tests/integration/test_integration.py

```diff
@@ -50,50 +50,61 @@ def get_experiment_status(path: str) -> str:
         "BLEU",
     ],
 )
-def test_oasst_training(tmp_path, config_name, metric):
+def test_oasst_training_gpu(tmp_path, config_name, metric):
+    run_oasst(tmp_path, config_name, metric)
+
+
+@has_no_gpus
+@pytest.mark.parametrize(
+    "config_name",
+    [
+        "test_causal_language_modeling_oasst_cpu_cfg",
+    ],
+)
+@pytest.mark.parametrize(
+    "metric",
+    [
+        "Perplexity",
+        "BLEU",
+    ],
+)
+def test_oasst_training_cpu(tmp_path, config_name, metric):
+    run_oasst(tmp_path, config_name, metric)
+
+
+def run_oasst(tmp_path, config_name, metric):
     """
     Test training on OASST dataset.
 
     Pytest keeps around the last 3 test runs in the tmp_path fixture.
     """
-
     prepare_default_dataset(tmp_path)
-
     train_path = os.path.join(tmp_path, "train_full.pq")
-
-    config_path = (
-        Path.cwd() / "tests" / "src" / "integration" / f"{config_name}.yaml"
-    ).resolve()
-    with open(config_path, "r") as fp:
+    with open(
+        os.path.join(
+            os.path.dirname(os.path.realpath(__file__)), f"{config_name}.yaml"
```

Comment:
Is there a reason to have unit tests in `src` and the integration tests in the parent dir?

Reply:
The idea to move integration tests out of src was to make them more explicit and have tests in src be unit tests for the respective src folder.